agent/agent.go

```diff
@@ -160,6 +161,15 @@ type notifier interface {
 // mode, it runs a full Consul server. In client-only mode, it only forwards
 // requests to other Consul servers.
 type Agent struct {
+
+	// autoConfigJSON is the JSON serialized auto-config configuration to be
+	// used when reloading a configuration
+	autoConfigJSON string
+
+	// flags are needed when rebuilding our configuration after an auto-config
+	// update
+	configFlags config.Flags
```

Comment:
Could we create a new struct type for all the new methods that are being added to `Agent` and make these fields on the new struct? `Agent` is already so large it feels like we are hurting ourselves if we add anything more to it.

Reply:
The agent is a mess. Especially the interaction with the command package too. I am running into an issue with some test failures and having to rethink how this all fits together.

Reply:
I ended up putting it in a autoconf sub-package in the agent package. I am much happier with the structure of things after the refactor now.